packages/compass-indexes/src/modules/search-indexes.ts

```diff
@@ -505,29 +501,52 @@ export const fetchSearchIndexes = (): IndexesThunkAction<
       return;
     }
 
-    if (!dataService || !dataService.isConnected()) {
-      debug('warning: trying to load indexes but dataService is disconnected');
+    const notFetchableStatuses: SearchIndexesStatus[] = [
+      'NOT_AVAILABLE',
+      'FETCHING',
+      'POLLING',
+      'REFRESHING',
+    ];
+    // If we are currently doing fetching indexes, we will
+    // wait for that
+    if (notFetchableStatuses.indexOf(status) !== -1) {
```

Comment:
Picky, but we can use notFetchableStatuses.includes(status).

Reply:
added in [13d92db](https://github.com/mongodb-js/compass/pull/4883/commits/13d92db4ddd4b56db150eea6594cb8febdc1182c)